lib/puppet/type/cisco_bfd_global.rb

```diff
@@ -0,0 +1,294 @@
+# Manages the Cisco Bfd Global configuration resource.
+#
+# May 2016
+#
+# Copyright (c) 2016 Cisco and/or its affiliates.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+Puppet::Type.newtype(:cisco_bfd_global) do
+  @doc = "
+    Manages the Cisco Bfd Global configuration resource.
+    cisco_bfd_global {'default':
+      ..attributes..
+    }
+    'default' is only acceptable name for this global config object.
+    Example:
+    cisco_bfd_global { 'default':
+      ensure                => 'present',
+      echo_interface        => 'loopback10',
+      echo_rx_interval      => 300,
+      fabricpath_interval   => ['750', '350', '45'],
+      fabricpath_slow_timer => 15000,
+      fabricpath_vlan       => 100,
+      interval              => ['100', '100', '25'],
+      ipv4_echo_rx_interval => 100,
+      ipv4_interval         => ['200', '200', '50'],
+      ipv4_slow_timer       => 10000,
+      ipv6_echo_rx_interval => 200,
+      ipv6_interval         => ['500', '500', '30'],
+      ipv6_slow_timer       => 25000,
+      slow_timer            => 5000,
+      startup_timer         => 25,
+    }
+  "
+
+  ensurable
+
+  ###################
+  # Resource Naming #
+  ###################
+
+  newparam(:name, namevar: :true) do
+    desc "Instance of bfd global, only allow the value 'default'"
+
+    validate do |name|
+      if name != 'default'
+        error "only 'default' is accepted as a valid bfd_global name"
+      end
+    end
+  end # param id
+
+  ##############
+  # Attributes #
+  ##############
+
+  newproperty(:echo_interface) do
+    desc "Loopback interface used for echo frames. Valid values are
+          string, keyword 'default'."
+
+    munge do |value|
+      value = :default if value == 'default'
+      value
+    end
+  end # property echo_interface
+
+  newproperty(:echo_rx_interval) do
+    desc "Echo receive interval in msec. Valid values are
+          integer, keyword 'default'."
+
+    munge do |value|
+      value = :default if value == 'default'
+      begin
+        value = Integer(value) unless value == :default
+      rescue
+        raise 'echo_rx_interval must be a valid integer, or default.'
+      end
+      value
+    end
```

Comment:
This munge is a little messy and rescues should be avoided when possible. 

Note that if `Integer()` is passed a non-integer it will raise a meaningful error, so you can just do this instead:

```
    munge { |value| value == 'default' ? :default : Integer(value) }
```

*side note: these integer properties are sometimes easier to deal with if you just treat them all as strings. But I wouldn't change this at this point.*